tests/pos/i16407.scala

```diff
@@ -0,0 +1,7 @@
+trait X { //missing requirement: self type Z[?] & X of trait X does not conform to self type Z[X.this.A] of required trait Z
+  self: Z[_] =>
+}
+
+trait Z[A] extends X {
+  self: Z[A] => // comment this to compile successfully
```

Comment:
The comments in this `pos` file are not accurate. They say there's an error but clearly there isn't.